File: backend/src/main/scala/domains/blog/application/output/BlogCommentSummary.scala
package domains.blog.application.output

import domains.blog.model.*

import domains.user.model.UserIdentity

import java.time.Instant

final case class BlogCommentSummary(
  id: BlogCommentId,
  parentId: Option[BlogCommentId],
  content: BlogCommentContent,
  author: UserIdentity,
  score: Int,
  viewerVote: Option[BlogVote],
  createdAt: Instant,
  updatedAt: Instant
)

File: frontend/src/features/blog/http/response/BlogCommentSummary.ts
import type { UserIdentity } from '@/features/user/domain/user'
import type { BlogCommentContent } from '@/features/blog/model/BlogCommentContent'
import type { BlogCommentId } from '@/features/blog/model/BlogCommentId'
import type { BlogVote } from '@/features/blog/model/BlogVote'

export type BlogCommentSummary = {
  id: BlogCommentId
  parentId: BlogCommentId | null
  content: BlogCommentContent
  author: UserIdentity
  score: number
  viewerVote: BlogVote | null
  createdAt: string
  updatedAt: string
}
